src/cascadia/TerminalSettingsEditor/Profiles.h

```diff
@@ -85,11 +85,25 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
     struct ProfilePageNavigationState : ProfilePageNavigationStateT<ProfilePageNavigationState>
     {
     public:
-        ProfilePageNavigationState(const Editor::ProfileViewModel& viewModel, const Windows::Foundation::Collections::IMapView<hstring, Model::ColorScheme>& schemes, const IHostedInWindow& windowRoot) :
+        ProfilePageNavigationState(const Editor::ProfileViewModel& viewModel,
+                                   const Windows::Foundation::Collections::IMapView<hstring, Model::ColorScheme>& schemes,
+                                   const Editor::ProfilePageNavigationState& lastState,
+                                   const IHostedInWindow& windowRoot) :
             _Profile{ viewModel },
             _Schemes{ schemes },
             _WindowRoot{ windowRoot }
         {
+            // If there was a previous nav state, and it was for the same
+            // profile, then copy the selected pivot from it.
+            if (lastState)
+            {
+                const auto& oldGuid = lastState.Profile().Guid();
+                const auto& newGuid = _Profile.Guid();
+                if (oldGuid == newGuid)
```

Comment:
I want to see if this works after #8837 merges since the guid may be autogenerated based on the name. This isn't your fault. We check the guid in other places too so this is bound to be a common problem.

Reply:
Yea that'll definitely happen if you rename a profile that doesn't otherwise have a guid. We could theoretically manually set the GUID to the generated one, though I dunno if that's more annoying or not. 

Or I suppose if we rename a profile, we could generate the new GUID, then slap that into the _lastProfileNavState.Profile().Guid(), but I guess TSM is the thing that knows about GUID generation. That's not a public interface. Hmmm.